### DISCUSSION TITLE
Support nodejs worker_threads

### DISCUSSION THREAD

**Discussion by TimDaub** [mswjs/msw#1205] [Ideas]
### Scope

Improves an existing behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

In Discord, I asked:

> Hey I'm using msw in nodejs. I'm doing fetch requests in a worker_thread, will msw intercept those?

The response was:

> Hey, @tim . Currently no. We have support for interception in a child process but it's not a part of MSW yet, as far as I know.

So this issue is documenting that worker_threads aren't supported currently.

Details that could help during implementation:

> Regarding child process interception, you can read how it functions right now in here: https://github.com/mswjs/interceptors#remotehttpinterceptor
> Shortly: you establish an interceptor in a child process, and a remote resolver in a parent process. Then the library uses IPC to message requests/responses between the processes.

**Comment by kettanaito** [comment]
Hi, @TimDaub. Thanks for this proposal. 

That response appears to be correct. We support request interception via IPC in the low-level Interceptors library but not as a part of MSW (yet). If you wish to enable API mocking in your worker, you would have to follow the [Node.js integration](https://mswjs.io/docs/getting-started/integrate/node) and set it up in that worker. You won't have the means to affect the library's behavior from a parent process at the moment. 

We are exploring a better experience around this but this has a low at the moment.
